x-pack/plugins/actions/server/lib/is_conector_deprecated.ts

```diff
@@ -18,11 +18,39 @@ export const isConnectorDeprecated = (
      * Connectors after the Elastic ServiceNow application use the
```

Comment:
@gmmorris @mikecote @pmuellr @ymao1 I changed the logic of the function because it seemed wrong to me. Let me know if my assumptions are wrong. I put comments on each decision.

Reply:
IIRC the original implementation was a copy-paste of the original front end code.
You're far more familiar with ServiceNow than I am, so I trust you to change this check as you see fit.

Reply:
Thanks! That was probably a mistake on my part because I was not aware of the pre-configured connectors.

Reply:
I guess this implies `connector.config`, at least for pre-configured connectors, has not already been passed through the validators, and so wouldn't be set to `true` because it wasn't there.  It would get passed through the validators by the time it was executed though.

Wonder if this will cause us a problem in the future, if we start validating that config before associating it with the connector.

Reply:
FWIW, I `console.log` inside the function with pre-configured connectors and non pre-configured connectors and the config was never null.

Reply:
@cnasikas I had a case I tested with @peteharverson where the config was null.

Reply:
@academo can you verify this edge case with @pmuellr please?
My understanding was that such a case shouldn't happen as this method is only called on existing connectors, all of whom should have a config.
Something might be wrong if you're seeing that happen.

Reply:
@academo Do you mean on the front end?

Reply:
I think the best is try with Peter. I could not replicate this case myself and I wonder if it was a case of an old elastic without migrations or a wrong configuration.

Reply:
I think you are confusing the backend with the frontend. I think that @peteharverson is not aware of what is going on in this function (backend).

Reply:
indeed, the API still omits the configuration when returned to the front end.
This is why we moved the check to the back end and added the isDeprecated flag so that you won't need the config on the front end.

Reply:
you are correct! that's the case I was mentioning. Excuses for the confusion.

Reply:
No worries :)